rasp/tor_engine.py

```diff
@@ -44,29 +45,13 @@ def send_signal(self):
 
     def get_page_source(self, url):
         try:
-            if url:
-                self.send_signal()
-                req = urllib.request.Request(url, self.data, self.headers)
-                res = urllib.request.urlopen(req)
-                if res:
-                    try:
-                        return Webpage(url, str(res.read()))
-                    except http.client.IncompleteRead:
-                        return None
-                else:
-                    return None
-            else:
-                return None
+            self.send_signal()
```

Comment:
Not related: this sends a `NEWNYM` signal to the control port on every `GET`. That is probably less than ideal in some cases. In the use case we created this library for, it was needed, but for many others this would be less than ideal.